frontend/src/tests/components/OurTable.test.js

```diff
@@ -1,674 +1,432 @@
-        var tester = true;
-        try {
-            await screen.findByTestId("testid-back-two-page-button");
-            tester = false;
-        } catch(e) { }
-        expect(tester).toBe(true);
-        fireEvent.click(nextButton);
-        expect(await screen.findByTestId("testid-current-page-button")).toContainHTML("3");
-        expect(await screen.findByTestId("testid-back-two-page-button")).toBeInTheDocument();
-    });
-
-    test("renders a table with 100 rows and tests the last page button visibiliy", async () => {
-        render(
-            <OurTable columns={columns} data={hundredRows} />
-        );
-
-        expect(await screen.findByTestId("testid-last-page-button")).toContainHTML("10");
-        const lastButton = screen.getByTestId("testid-last-page-button");
-        fireEvent.click(lastButton);
-        expect(await screen.findByTestId("testid-current-page-button")).toContainHTML("10");
-        expect(await screen.findByTestId("testid-back-one-page-button")).toContainHTML("9");
-        const backOneButton = screen.getByTestId("testid-back-one-page-button");
-        fireEvent.click(backOneButton);
-        expect(await screen.findByTestId("testid-current-page-button")).toContainHTML("9");
-        expect(await screen.findByTestId("testid-forward-one-page-button")).toContainHTML("10");
-        expect(lastButton).not.toBeInTheDocument();
-        fireEvent.click(backOneButton);
-        expect(await screen.findByTestId("testid-current-page-button")).toContainHTML("8");
-        expect(lastButton).not.toBeInTheDocument();
-        fireEvent.click(backOneButton);
-        expect(await screen.findByTestId("testid-current-page-button")).toContainHTML("7");
-        expect(lastButton).not.toBeInTheDocument();
+  }
+
+  const fiftyOneRows = [];
+  for (let i = 1; i <= 51; i++) {
+    fiftyOneRows.push({
+      col1: `Hello ${i}`,
+      col2: `World ${i}`,
+      createdAt: `2021-04-01T04:00:00.000`,
+      log: `foo\nbar\n  baz ${i}`,
     });
+  }
+
+  const clickMeCallback = jest.fn();
+
+  const columns = [
+    {
+      Header: "Column 1",
+      accessor: "col1", // accessor is the "key" in the data
+    },
+    {
+      Header: "Column 2",
+      accessor: "col2",
+    },
+    ButtonColumn("Click", "primary", clickMeCallback, "testId"),
+    DateColumn("Date", (cell) => cell.row.original.createdAt),
+    PlaintextColumn("Log", (cell) => cell.row.original.log),
+  ];
+
+  test("The button appears in the table", async () => {
+    render(<OurTable columns={columns} data={threeRows} />);
+
+    expect(
+      await screen.findByTestId("testId-cell-row-0-col-Click-button")
+    ).toBeInTheDocument();
+    const button = screen.getByTestId("testId-cell-row-0-col-Click-button");
+    fireEvent.click(button);
+    await waitFor(() => expect(clickMeCallback).toBeCalledTimes(1));
+  });
+
+  test("default testid is testId", async () => {
```

Comment:
This was a test I had left unchanged from the original code. It does seem unnecessary but at the same time this test also would prevent someone from changing the "testid" in `OurTable.js` since every single test relies on testId being, well, `testid`